lib/client-api/src/client_api.ts

```diff
@@ -82,23 +82,45 @@ const withSubscriptionTracking = (storyFn: StoryFn) => {
   return result;
 };
 
+let _globalDecorators: DecoratorFunction[] = [];
+
+let _globalParameters: Parameters = {};
+
+/**
+ * Add a global decorator function to be applied to all stories
+ * within the given framework, or to all stories if no framework
+ * is specified.
+ */
+export const addDecorator = (decoratorFn: DecoratorFunction, framework?: string) => {
+  // ignore framework for now, but add it to the API for the future
+  _globalDecorators.push(decoratorFn);
+};
+
+export const addParameters = (parameters: Parameters) => {
```

Comment:
Should this have `framework` too?

Reply:
Removed the framework arg entirely in favor of using `parameters.framework`, which is now available and simplifies the API.

Reply:
So the decorator should just check `parameters.framework` and decide whether to apply itself?

Reply:
Yeah. It's a little tricky, but I removed it because getting the right framework to add in the context of presets, zero-config, turns out also to be tricky. At least with the decorator, we will be guaranteed to have the framework.

I'm thinking more of the following:

```
import centered from '@storybook/addon-centered"

addDecorator(centered)
```

And centered would switch on the `framework` to pick which version to use. (Or do nothing if the framework isn't supported, or potentially throw if that was more appropriate).